File: CaptchaForge/src/main/java/com/cunoc/CaptchaForge/Model/Analyzer/ReportErrorInterpreter.java
package com.cunoc.CaptchaForge.Model.Analyzer;

public class ReportErrorInterpreter {
    private ErrorTypeInTheInterpreter type;
    private Token toke;
    private String description;

    public ReportErrorInterpreter(ErrorTypeInTheInterpreter type, Token toke, String description) {
        this.type = type;
        this.toke = toke;
        this.description = description;
    }

    public ErrorTypeInTheInterpreter getType() {
        return type;
    }

    public Token getToke() {
        return toke;
    }

    public String getDescription() {
        return description;
    }

    @Override
    public String toString() {
        return "- Tipo de error:" + type.toString() + "\n - Token afectado: " + toke.toString() + "\n - Descripción: "
                + description.toString();
    }
}

File: Angular/src/app/model/Analyzer/ReportErrorInterpreter.ts
import { ErrorTypeInTheInterpreter } from "./ErrorTypeInTheInterpreter";
import { Token } from "./Token";

export class ReportErrorInterpreter {
    private type: ErrorTypeInTheInterpreter;
    private toke: Token;
    private description: string;

    constructor(type: ErrorTypeInTheInterpreter, toke: Token, description: string) {
        this.type = type;
        this.toke = toke;
        this.description = description;
    }

    public getType(): ErrorTypeInTheInterpreter {
        return this.type;
    }

    public getToke(): Token {
        return this.toke;
    }

    public getDescription(): string {
        return this.description;
    }

    public toString(): string {
        return `- Tipo de error: ${this.type}\n - Token afectado: ${this.toke.toString()}\n - Descripción: ${this.description}`;
    }

    public static fromJSON(data: any): ReportErrorInterpreter {
        return new ReportErrorInterpreter(
            data.type || '',
            Token.fromJSON(data.toke),
            data.description || ''
        );
    }
}